internal/integrations/v4/runner/runner.go

```diff
@@ -299,16 +300,38 @@ func (r *runner) handleStderr(stderr <-chan []byte) {
 		// obfuscated stderr
 		obfuscatedLine := helpers.ObfuscateSensitiveDataFromString(string(line))
 
-		if r.log.IsDebugEnabled() {
-			r.log.WithField("line", obfuscatedLine).Debug("Integration stderr (not parsed).")
-		} else {
-			fields := r.stderrParser(obfuscatedLine)
-			if v, ok := fields["level"]; ok && (v == "error" || v == "fatal") {
+		var lineWasParsed bool
+
+		for _, errParser := range r.stderrParser {
+			fields := errParser(obfuscatedLine)
+			if v, ok := fields["level"]; ok {
 				// If a field already exists, like the time, logrus automatically adds the prefix "deps." to the
 				// Duplicated keys
-				r.log.WithFields(logrus.Fields(fields)).Info("received an integration log line")
+
+				logLine := r.log.WithFields(logrus.Fields(fields))
+				logMessage := "integration log"
+
+				switch strings.ToLower(v.(string)) {
+				case "info":
+					logLine.Info(logMessage)
+				case "debug":
+					logLine.Debug(logMessage)
+				case "trace":
+					logLine.Trace(logMessage)
+				case "warn", "warning":
+					logLine.Warn(logMessage)
+				default:
+					logLine.Error(logMessage)
+				}
+
+				lineWasParsed = true
+				break
 			}
 		}
+
+		if r.log.IsDebugEnabled() && !lineWasParsed {
```

Comment:
Is this `IsDebugEnabled` needed? I mean, if we log as `debug` inside the `if` then it will be printed only when we configure the `debug` log level. Or is this for something else?

Reply:
mmm..... yes right fixed

Reply:
If I understood you well, I think the idea is to log it only when in debug and when the line wasn't parsed. So it shouldn't fit the case of `debug` inside the if 🤔

Reply:
I might be missing something, but if the idea is to log it only when in debug, that is already achieved by the call to `.Debug()` inside the if, no? I mean, without the `IsDebugEnabled()` the call will be done in all cases but the log line will only appear when having configured the debug log level. Not sure if that (subtle?) difference is needed here.